ccm_web/client/src/App.tsx

```diff
@@ -62,6 +64,12 @@ function App (): JSX.Element {
     )
   }
 
+  if (!globals.user.hasCanvasToken) {
+    // Initiate OAuth flow
+    location.href = '/canvas/redirectOAuth'
```

Comment:
I did some reading with `location.href.` There are quiet a few option to do a redirect like `assign`, `replace`. This [article](https://marikalam.medium.com/how-do-you-redirect-to-another-page-in-javascript-6c7524c1f88a) suggest `replace` is best as it doesn't keep the browser history.

like this `location.replace(/canvas/redirectOAuth)`

Reply:
Hm, interesting. I did see this, but I wasn't sure whether we should remove the history. Maybe we don't want them to be able to go back in this instance. I'll take a look in a minute.

Reply:
I did some testing; this doesn't seem to be a huge difference. I did notice that the if the user hits the back button with `location.replace`, eventually they will get a `404`. This doesn't happen with `location.href`, it just seems to keep sending them to the application root. I can't explain that, but I think the latter is preferred, so I'm inclined to keep it as is.